### PR TITLE
Add `max_trials` scheduler config parameter

### PR DESCRIPTION
**Pull request by motus**
maybe merge after #713 (they might conflict)

Part of issue #715

### CODE REVIEW

File: mlos_bench/mlos_bench/config/schemas/schedulers/scheduler-schema.json [hunk 1]
```diff
@@ -25,6 +25,12 @@
                     "description": "Whether to teardown the experiment after running it.",
                     "type": "boolean"
                 },
+                "max_trials": {
+                    "description": "Max. number of trials to run",
+                    "type": "integer",
+                    "minimum": 1,
```
Comment (bpkroth) [inline mlos_bench/mlos_bench/config/schemas/schedulers/scheduler-schema.json:31]: Should allow this to support -1 as well and document what that means.

File: mlos_bench/mlos_bench/schedulers/base_scheduler.py [hunk 2]
```diff
@@ -234,10 +236,20 @@ def _run_schedule(self, running: bool = False) -> None:
         for trial in self.experiment.pending_trials(datetime.utcnow(), running=running):
             self.run_trial(trial)
 
+    def not_converged(self) -> bool:
```
Comment (bpkroth) [inline mlos_bench/mlos_bench/schedulers/base_scheduler.py:239]: May want to consider renaming this to "not_done" or something like that.

Converged means something specific in the optimizer case, but doesn't quite have the same meaning here.

Comment (motus) [reply mlos_bench/mlos_bench/schedulers/base_scheduler.py]: fixed

File: mlos_bench/mlos_bench/config/schedulers/sync_scheduler.jsonc [hunk 3]
```diff
@@ -6,6 +6,7 @@
 
     "config": {
         "trial_config_repeat_count": 3,
+        "max_trials": 100,
```
Comment (bpkroth) [inline mlos_bench/mlos_bench/config/schedulers/sync_scheduler.jsonc:9]: Set the default to -1?

**Review by bpkroth — APPROVED** [review]

**Pull request merged**